### ISSUE TITLE
NotIn where querymod

### ISSUE THREAD

**Issue by ImVexed** [volatiletech/sqlboiler#638]
Similarly to: https://stackoverflow.com/questions/56518744/sqlboiler-not-in-clause

I'm trying to use the `models.{type}Where.{field}.IN` querymod, but instead of it being inclusive I want it to be exclusive. I too a look around `qm` as well and found https://godoc.org/github.com/volatiletech/sqlboiler/queries/qm#WhereIn but not `WhereNotIn` 

Am I missing something? Is there a way to negate a query mod?

**Comment by nadilas** [comment]
Hi @ImVexed, dependent on your sql driver you are using. @aarondl correct me if I'm wrong, but I'm guessing the lack of `WhereNotIn` has something to do with its implications. 
I'd advise to use a `NOT EXISTS` or an `EXCEPT` clause wherever possible. 
Here's a nice summary on the topic: https://www.red-gate.com/hub/product-learning/sql-prompt/consider-using-not-exists-instead-not-subquery

TL;DR: `NOT IN` doesn't play nice when `NULL` values are possible

**Comment by aarondl** [comment]
There is no current way to negate a query mod. This is mostly because of how these Wherein helpers grew somewhat organically. Although it doesn't sound like we could implement it with much success.

Originally where queries were specified as strings with no constants and so we never had to negate a query mod since a user would simply write the negated clause. It's also true in the query mod system it'd be a bit strange given that a negation that conformed to the interface could negate a select query too. What would that mean?

It may be able to be generated correctly though. 

Thank you for your reply @nadilas. It was informative.

**Comment by aarondl** [comment]
These exist now (awkwardly enough)

**Comment by istrau2** [comment]
@aarondl I can't find `models.{type}Where.{field}.IN` or `models.{type}Where.{field}.NIN` in my models. Why would that be?

**Comment by aarondl** [comment]
@istrau2 These helpers are only for Go primitives in general. Simple things like int64s and what-not. There is code preventing them being generated for more complicated things.
